dask/array/reshape.py

```diff
@@ -226,6 +231,36 @@ def reshape(x, shape, merge_chunks=True):
         x = x.rechunk({i: 1 for i in range(din - dout)})
 
     inchunks, outchunks = reshape_rechunk(x.shape, shape, x.chunks)
+    # Check output chunks are not too large
+    max_chunksize_in_bytes = np.prod([max(i) for i in outchunks]) * x.dtype.itemsize
+    if (
+        limit is None
+        and config.config["array"]["slicing"]["split-large-chunks"] is not False
+    ):
+        limit = parse_bytes(config.get("array.chunk-size"))
+    if max_chunksize_in_bytes > limit:
+        # Leave chunk sizes unaltered where possible
+        matching_chunks = Counter(inchunks) & Counter(outchunks)
```

Comment:
This would fail for something like
```
inchunks = ((1,1,1), (2,2,2))
outchunks = ((2,2,2), (3,3,3))

from collections import Counter

Counter(inchunks) & Counter(outchunks)
# Counter({(2, 2, 2): 1})
```

but I can't think of an array example that would result in a similar situation.

Reply:
I think that is the desired behaviour in those circumstances? I agree, this would be easier to talk about if we had an array example.

General rule is: if you have output chunks that look like they match chunks from an input dimension, preserve the chunk structure, otherwise use "auto" chunking.